### PR TITLE
revise: getters

### PR DESCRIPTION
**Pull request by MohammedTaka**

### CODE REVIEW

File: contracts/facets/GettersFacet.sol [hunk 1]
```diff
@@ -22,6 +22,169 @@ import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet
 contract GettersFacet is IGetters {
     using EnumerableSet for EnumerableSet.AddressSet;
 
+    /// @notice This function is used as a helper for front-end implementation
+    /// @param termId The term id for which the summary is being requested
+    /// @return term complete term object
+    /// @param joinPositions available positions if any
+    /// @param joinAmounts the minimum security deposit for each position from joinPositions
+    function getTurnGroupRelatedSummary(
```
Comment (MohammedTaka) [inline contracts/facets/GettersFacet.sol]: Missing natspec of the other return variables

File: contracts/facets/GettersFacet.sol [hunk 2]
```diff
@@ -22,6 +22,169 @@ import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet
...
+        remainingCycles = getRemainingCycles(termId);
+
+        // Contributions and prices
+        contributionsAndPrices = [getRemainingCyclesContributionWei(termId), getLatestPrice()];
+
+        // Collateral values
+        collateralValues = [collateral.firstDepositTime, collateral.counterMembers];
+
+        // Fund values
+        fundValues = [
```
Comment (MohammedTaka) [inline contracts/facets/GettersFacet.sol]: Please also add state values like the fund and collateral state

File: contracts/facets/GettersFacet.sol [hunk 3]
```diff
@@ -22,6 +22,169 @@ import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet
...
+    /// @param termId The term id for which the summary is being requested
+    /// @return term complete term object
+    /// @param joinPositions available positions if any
+    /// @param joinAmounts the minimum security deposit for each position from joinPositions
+    function getTurnGroupRelatedSummary(
+        uint termId
+    )
+        external
+        view
+        returns (
```
Comment (MohammedTaka) [inline contracts/facets/GettersFacet.sol:46]: Couldn't we make structs for this inside the getters facet instead? Specific structs for this that would contain the return values of each summary function. For example the struct of the fund would contain:

            fund.initialized,
            fund.currentState,
            fund.stableToken,
            fund.beneficiariesOrder,
            fund.fundStart,
            fund.fundEnd,
            fund.currentCycle,
            fund.totalAmountOfCycles

File: contracts/interfaces/IGetters.sol [hunk 4]
```diff
@@ -7,8 +7,64 @@ import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
...
+    ///                                      cycles contribution in wei, latest price from Chainlink
+    /// @return collateralRelated A helper struct with the following values:
+    ///                           collateral first deposit time in seconds, collateral counter members
+    /// @return fundRelated A helper struct with the following values:
+    ///                     fund start time in seconds, fund end time in seconds, current cycle, expelled
+    ///                     participants, total amount of cycles
+    /// @return yieldRelated A helper struct with the following values:
+    ///                      yield start time in seconds, total deposit in wei, current total deposit in wei,
+    ///                      total shares
+    function getTurnGroupRelatedSummary(
```
Comment (MohammedTaka) [inline contracts/interfaces/IGetters.sol]: Rename to Term, we do not use Turn Group in the smart contracts

**Review by elallamTaka — APPROVED** [review]

**Pull request merged**